e2e/support/commands/api/question.ts

```diff
@@ -0,0 +1,239 @@
+   */
+  type?: Card["type"];
+  /**
+   * Defaults to "table".
+   */
+  display?: Card["display"];
+  parameters?: Card["parameters"];
+  visualization_settings?: Card["visualization_settings"];
+  /**
+   * Parent collection in which to store this question.
+   */
+  collection_id?: Card["collection_id"];
+  /**
+   * Used on the frontend to determine whether the question is pinned or not.
+   */
+  collection_position?: Card["collection_position"];
+  embedding_params?: Card["embedding_params"];
+  /**
+   * Defaults to false.
+   */
+  enable_embedding?: Card["enable_embedding"];
+};
+
+type StructuredQuestionDetails = Omit<QuestionDetails, "dataset_query"> & {
+  /**
+   * Defaults to SAMPLE_DB_ID.
+   */
+  database?: DatasetQuery["database"];
+  query: StructuredQuery;
+};
+
+type NativeQuestionDetails = Omit<QuestionDetails, "dataset_query"> & {
+  /**
+   * Defaults to SAMPLE_DB_ID.
+   */
+  database?: DatasetQuery["database"];
+  native: NativeQuery;
+};
+
+type Options = {
+  /**
+   * Whether to visit the question in order to load its metadata.
+   * Defaults to false.
+   */
+  loadMetadata?: boolean;
+  /**
+   * Whether to visit the question after the creation or not.
+   * Defaults to false.
+   */
+  visitQuestion?: boolean;
+  /**
+   * Whether to wrap a question id, to make it available outside of this scope.
+   * Defaults to false.
+   */
+  wrapId?: boolean;
+  /**
+   * Alias a question id in order to use it later with `cy.get("@" + alias).
+   * Defaults to "questionId".
+   */
+  idAlias?: string;
+  /**
+   * We need distinctive endpoint aliases for cases where we have multiple questions or nested questions.
+   * Defaults to "cardQuery".
+   */
+  interceptAlias?: string;
+};
+
+Cypress.Commands.add("archiveQuestion", (id: Card["id"]) => {
+  cy.log(`Archiving a question with id: ${id}`);
+
+  return cy.request("PUT", `/api/card/${id}`, {
+    archived: true,
+  });
+});
+
+Cypress.Commands.add(
+  "createQuestion",
+  (questionDetails: StructuredQuestionDetails, options?: Options) => {
+    const { database = SAMPLE_DB_ID, name, query } = questionDetails;
+
+    if (!query) {
+      throw new Error('"query" attribute missing in questionDetails');
+    }
+
+    logAction("Create a QB question", name);
+
+    return question(
+      {
+        ...questionDetails,
+        dataset_query: { type: "query", query, database },
+      },
+      options,
+    );
+  },
+);
+
+Cypress.Commands.add(
+  "createNativeQuestion",
+  (questionDetails: NativeQuestionDetails, options?: Options) => {
+    const { database = SAMPLE_DB_ID, name, native } = questionDetails;
+
+    if (!native) {
+      throw new Error('"native" attribute missing in questionDetails');
+    }
+
+    logAction("Create a native question", name);
+
+    return question(
+      {
+        ...questionDetails,
+        dataset_query: { type: "native", native, database },
+      },
+      options,
+    );
+  },
+);
+
+const question = (
```

Comment:
I simplified this function by removing the `queryType` argument, `datatabase` property, and the conditional `native` vs `query` property picking, by replacing them with `dataset_query` property.

Interface of `cy.createQuestion` and `cy.createNativeQuestion` has not changed.